lib/lotus/model/adapters/memory/collection.rb

```diff
@@ -77,9 +84,18 @@ def initialize(name, identity)
           # @api private
           # @since 0.1.0
           def create(entity)
-            @primary_key.increment! do |id|
-              entity[identity] = id
-              records[id] = entity
+            if _id = entity[identity]
+              if records[_id]
+                raise UniqueConstraintViolation
```

Comment:
what do you think about adding messages to help user identify which entity fail this constraint?